### PR TITLE
TD-55 Library Type Options List UI

### PR DESCRIPTION
**Pull request by arvinmybusybee**
TD-55 Library Type Options List UI

### CODE REVIEW

File: src/app/core/services/portal.service.ts [hunk 1]
```diff
@@ -106,4 +107,29 @@ export class PortalService {
   getCostCenters(): Observable<ICostCenter[]> {
     return this._portalApi.getCostCenters();
   }
+
+  getLibraryTypeOptions(): Observable<ILibraryTypeOption[]> {
+    return this._portalApi.getLibraryTypeOptions();
+  }
+  getLibraryTypeOption(id: number): Observable<ILibraryTypeOption> {
```
Comment (vernienamca) [inline src/app/core/services/portal.service.ts:114]: Please add spacing please between code. Do the same to others.

File: src/app/core/services/portal.service.ts [hunk 2]
```diff
@@ -106,4 +107,29 @@ export class PortalService {
...
+    return this._portalApi.getLibraryTypeOption(id);
+  }
+  updateLibraryTypeOption(data: any): Observable<any> {
+    return this._portalApi.updateLibraryTypeOption(data);
+  }
+  exportLibraryTypeOptions(): void {
```
Comment (vernienamca) [inline src/app/core/services/portal.service.ts:120]: Please add spacing please between code. Do the same to others.

File: src/app/pages/apps/library-type-option/library-type-option.component.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    private _portalService: PortalService,
+    private _snackBar: MatSnackBar
+  ) {
+    this.form = this._fb.group({
+      type: ['', [Validators.required]],
+      code: ['', [Validators.required]],
+      description: ['', [Validators.required]],
+      status: ['', [Validators.required]],
+      remarks: ['']
+      // ,isActive: [true, []]
```
Comment (vernienamca) [inline src/app/pages/apps/library-type-option/library-type-option.component.ts:47]: Remove this commented code if not needed.

File: src/app/pages/apps/library-type-option/library-type-option.component.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          return;
+        }
+        this.form.setValue({
+          type: data.type,
+          code: data.code || '',
+          description: data.description || '',
+          status: data.status,
+          remarks: data.remarks
+          // ,isActive: data.isActive === 'Y'
+        });
```
Comment (vernienamca) [inline src/app/pages/apps/library-type-option/library-type-option.component.ts:64]: Remove this commented code if not needed.

File: src/app/pages/apps/library-type-option/library-type-option.module.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    MatTooltipModule,
+    ReactiveFormsModule,
+    MatSelectModule,
+    MatButtonToggleModule,
+    RelativeDateTimeModule,
+    MatProgressBarModule,
+    MatDialogModule,
+    MatDividerModule,
+    FlexLayoutModule,
+
```
Comment (vernienamca) [inline src/app/pages/apps/library-type-option/library-type-option.module.ts:54]: Remove the spacing.

**Review by vernienamca — CHANGES_REQUESTED** [review]

File: src/app/core/services/portal.service.ts (hunk 1 above)
Comment (arvinmybusybee) [reply src/app/core/services/portal.service.ts:114]: Kindly see update

File: src/app/core/services/portal.service.ts (hunk 2 above)
Comment (arvinmybusybee) [reply src/app/core/services/portal.service.ts:120]: Kindly see update

File: src/app/pages/apps/library-type-option/library-type-option.component.ts (hunk 3 above)
Comment (arvinmybusybee) [reply src/app/pages/apps/library-type-option/library-type-option.component.ts:47]: Kindly see update

File: src/app/pages/apps/library-type-option/library-type-option.component.ts (hunk 4 above)
Comment (arvinmybusybee) [reply src/app/pages/apps/library-type-option/library-type-option.component.ts:64]: Kindly see update

File: src/app/pages/apps/library-type-option/library-type-option.module.ts (hunk 5 above)
Comment (arvinmybusybee) [reply src/app/pages/apps/library-type-option/library-type-option.module.ts:54]: Kindly see update

File: src/app/pages/apps/library-type-option/library-type-option-list/library-type-option-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    //   .pipe(
+    //     takeUntil(this._onDestroy$),
+    //     finalize(() => this.isListLoading = false)
+    //   )
+    //   .subscribe(data => {
+    //     if (!data) {
+    //       return;
+    //     }
+    //     this.subject$.next(data);
+    //   });
```
Comment (ferlyncaps) [inline src/app/pages/apps/library-type-option/library-type-option-list/library-type-option-list.component.ts:93]: Can we removed this please if unused? :)

**Review by ferlyncaps — APPROVED** [review]

**Pull request merged**